src/System.IO.FileSystem.DriveInfo/tests/DriveInfo.Windows.Tests.cs

```diff
@@ -126,53 +130,66 @@ public void TestValidDiskSpaceProperties()
+
+            Assert.Throws<DriveNotFoundException>(() => { var invalid = invalidDrive.AvailableFreeSpace; });
+            Assert.Throws<DriveNotFoundException>(() => { var invalid = invalidDrive.DriveFormat; });
+            Assert.Equal(DriveType.NoRootDirectory, invalidDrive.DriveType);
+            Assert.False(invalidDrive.IsReady);
+            Assert.Equal(invalidDriveName + ":\\", invalidDrive.Name);
+            Assert.Equal(invalidDriveName + ":\\", invalidDrive.ToString());
+            Assert.Equal(invalidDriveName + ":\\", invalidDrive.RootDirectory.FullName);
+            Assert.Throws<DriveNotFoundException>(() => { var invalid = invalidDrive.TotalFreeSpace; });
+            Assert.Throws<DriveNotFoundException>(() => { var invalid = invalidDrive.TotalSize; });
+            Assert.Throws<DriveNotFoundException>(() => { var invalid = invalidDrive.VolumeLabel; });
+            Assert.Throws<DriveNotFoundException>(() => { invalidDrive.VolumeLabel = null; });
         }
 
-        [Fact, OuterLoop]
-        [ActiveIssue(1355)]
+        [Fact]
         [PlatformSpecific(PlatformID.Windows)]
-        public void TestVolumeLabel()
+        public void GetVolumeLabel_Returns_CorrectLabel()
         {
+            int serialNumber, maxFileNameLen, fileSystemFlags;
+            int volNameLen = 50;
+            int fileNameLen = 50;
+            StringBuilder volumeName = new StringBuilder(volNameLen);
+            StringBuilder fileSystemName = new StringBuilder(fileNameLen);
+
             // Get Volume Label - valid drive
             var validDrive = DriveInfo.GetDrives().Where(d => d.DriveType == DriveType.Fixed).First();
-            const int volNameLen = 50;
-            StringBuilder volumeName = new StringBuilder(volNameLen);
-            const int fileSystemNameLen = 50;
-            StringBuilder fileSystemName = new StringBuilder(fileSystemNameLen);
-            int serialNumber, maxFileNameLen, fileSystemFlags;
-            bool r = GetVolumeInformation(validDrive.Name, volumeName, volNameLen, out serialNumber, out maxFileNameLen, out fileSystemFlags, fileSystemName, fileSystemNameLen);
+            bool r = GetVolumeInformation(validDrive.Name, volumeName, volNameLen, out serialNumber, out maxFileNameLen, out fileSystemFlags, fileSystemName, fileNameLen);
             if (r)
             {
                 Assert.Equal(volumeName.ToString(), validDrive.VolumeLabel);
             }
+            else // if we can't compare the volumeName, we should at least check that getting it doesn't throw
+            {
+                var name = validDrive.VolumeLabel;
+            }
+        }
 
-            // Set Volume Label - valid drive
-            string newLabel = "DriveInfoTests";
-            string originalLabel = validDrive.VolumeLabel;
-
-            if (validDrive.IsReady)
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)]
+        public void SetVolumeLabel_Roundtrips()
```

Comment:
VolumeLabel setting like this does make me slightly nervous since we don't really know which volume we're messing with when we run the tests. The old test set it to a different name in a ```try``` and then set it back to the original in the corresponding ```finally``` - I switched it to this in the hopes that it will be more disaster-proof regardless of which volume is being modified.